### PR TITLE
Feature flagging

### PR DESCRIPTION
**Pull request by jackschultz**
Ability to flag a comment.

### CODE REVIEW

**Comment by jackschultz** [comment]
The requirements to flag a comment is the same as the ability to create.

File: lib/flyover_comments/authorization.rb [hunk 1]
```diff
@@ -21,5 +21,15 @@ def can_create_flyover_comment?(comment, user)
       end
     end
 
+    def can_flag_flyover_comment?(comment, user)
+      if Object.const_defined?("Pundit") && policy = Pundit.policy(user, comment)
+        policy.create?
+      elsif user.respond_to?(:can_flag_flyover_comment?)
+        user.can_flag_flyover_comment?(comment)
+      else
```
Comment (bcroesch) [inline lib/flyover_comments/authorization.rb:29]: Let's return `!user.nil?` from the else clause. I think `comment.user == user` was a sensible default for creating a comment, but only being able to flag your own comments probably doesn't make sense as the default behavior. For flags, I think a signed in user is probably a sensible default.

File: spec/lib/flyover_comments/authorization_spec.rb [hunk 2]
```diff
@@ -54,5 +54,29 @@ def user.can_create_flyover_comment?(comment)
       end
     end
 
+    describe "can_flag_flyover_comment?" do
+      let(:user){ Ident::User.new }
+      let(:comment){ double(user: user) }
+
+      it "calls can_flag_flyover_comment? on the passed user if the user defines it" do
+        def user.can_flag_flyover_comment?(comment)
```
Comment (bcroesch) [inline spec/lib/flyover_comments/authorization_spec.rb:62]: :+1: love this

File: spec/lib/flyover_comments/authorization_spec.rb [hunk 3]
```diff
@@ -54,5 +54,29 @@ def user.can_create_flyover_comment?(comment)
...
+          true
+        end
+
+        expect(user).to receive(:can_flag_flyover_comment?).with(comment)
+        Widget.new.can_flag_flyover_comment?(comment, user)
+      end
+
+      it "returns true if user#can_flag_flyover_comment? isn't defined and the comment's user is the passed user" do
```
Comment (bcroesch) [inline spec/lib/flyover_comments/authorization_spec.rb]: This test will need to get updated to match the behavior described above

File: spec/models/flyover_comments/flag_spec.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+module FlyoverComments
+  RSpec.describe Flag, :type => :model do
+    pending "add some examples to (or delete) #{__FILE__}"
```
Comment (bcroesch) [inline spec/models/flyover_comments/flag_spec.rb]: Delete the pending

File: app/models/flyover_comments/flag.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+module FlyoverComments
+  class Flag < ActiveRecord::Base
+    belongs_to :comment
+    belongs_to :user
```
Comment (bcroesch) [inline app/models/flyover_comments/flag.rb:4]: validates :comment_id and :user_id

File: db/migrate/20150806215036_create_flyover_comments_flags.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class CreateFlyoverCommentsFlags < ActiveRecord::Migration
+  def change
+    create_table :flyover_comments_flags do |t|
+      t.references :comment, index: true
+      t.references :user, index: true
+
+      t.timestamps null: false
+    end
+    add_foreign_key :flyover_comments_flags, :comments
+    add_foreign_key :flyover_comments_flags, :users
```
Comment (bcroesch) [inline db/migrate/20150806215036_create_flyover_comments_flags.rb]: Let's delete the foreign keys, so we don't force this on the host app

**Pull request merged**